### PR TITLE
Fixes #RHIROS-966 - CSV data aggregation

### PR DESCRIPTION
**Pull request by patilsuraj767**

### CODE REVIEW

File: cmd/aggregator.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+				if _, err := os.Stat(outputDir); os.IsNotExist(err) {
+					if err := os.MkdirAll("a/b/c/d", os.ModePerm); err != nil {
+						panic(err.Error())
+					}
+				}
+			} else {
+				outputDir, _ = os.Getwd()
+			}
+			f, err := os.Open(input_file)
+			if err != nil {
```
Comment (upadhyeammit) [inline cmd/aggregator.go:36]: Can we extract this if block which is repetitive as function and call?

File: cmd/aggregator.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+				panic(err.Error())
+			}
+			defer f.Close()
```
Comment (upadhyeammit) [inline cmd/aggregator.go:39]: Does this mean we keep the f open when there is error?

File: cmd/aggregator.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+
+	"github.com/go-gota/gota/dataframe"
+	"github.com/spf13/cobra"
+
+	"github.com/redhatinsights/ros-ocp-backend/internal/processor"
```
Comment (upadhyeammit) [inline cmd/aggregator.go:11]: Can this be converted to local import?
something like `internal/processor`?

File: cmd/aggregator.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+			csv := csv.NewReader(f)
+			records, err := csv.ReadAll()
+			if err != nil {
+				panic(err.Error())
+			}
+
+			df := dataframe.LoadRecords(records)
+			df = processor.Aggregate_data(df)
+			fileio, err := os.Create(outputDir + "/output.csv")
```
Comment (upadhyeammit) [inline cmd/aggregator.go:49]: Do you think we should have var for output.csv?

**Review by upadhyeammit — COMMENTED** [review]
With my limited knowledge added few comments, request to check.

File: cmd/aggregator.go (hunk 2 above)
Comment (patilsuraj767) [reply cmd/aggregator.go:39]: No, `f.Close` first checks if `f` is nil - https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/os/file_posix.go;l=21 and then perform the action.

File: cmd/aggregator.go (hunk 1 above)
Comment (patilsuraj767) [reply cmd/aggregator.go:36]: No, we can't. This is how exception handling done in idiomatic Go style.

File: cmd/aggregator.go (hunk 4 above)
Comment (patilsuraj767) [reply cmd/aggregator.go:49]: ok I will create var `outputFile := outputDir + "/output.csv"`

File: internal/processor/aggregator.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+	s := df.Rapply(func(s series.Series) series.Series {
+		owner_name := s.Elem(index_of_owner_name).String()
+		owner_kind := s.Elem(index_of_owner_kind).String()
+		workload := s.Elem(index_of_workload).String()
+		workload_type := s.Elem(index_of_workload_type).String()
+		if owner_kind == "ReplicaSet" && workload == "<none>" {
+			return series.Strings([]string{"replicaset", owner_name})
+		} else if owner_kind == "ReplicationController" && workload == "<none>" {
+			return series.Strings([]string{"replicationcontroller", owner_name})
```
Comment (saltgen) [inline internal/processor/aggregator.go:31]: Shall we use the WorkloadType enum values here?

File: internal/processor/kruize_api.go [hunk 6]
```diff
@@ -2,169 +2,108 @@ package processor
...
-		return errors.New(error_string)
+		return fmt.Errorf("an Error Occured %v", err)
 	}
 	q := req.URL.Query()
-	q.Add("experiment_name", params["experiment_name"])
-	q.Add("deployment_name", params["deployment_name"])
-	q.Add("namespace", params["namespace"])
+	q.Add("experiment_name", experiment)
+	req.URL.RawQuery = q.Encode()
 	res, err := client.Do(req)
```
Comment (saltgen) [inline internal/processor/kruize_api.go:104]: Is the intention here to have recommendations saved to DB post the Recommender system is introduced?

**Review by saltgen — APPROVED** [review]
@patilsuraj767 LGTM :+1: 
Just a suggestion and a question for my understanding.

File: internal/processor/aggregator.go (hunk 5 above)
Comment (patilsuraj767) [reply internal/processor/aggregator.go]: Done

File: internal/processor/kruize_api.go (hunk 6 above)
Comment (patilsuraj767) [reply internal/processor/kruize_api.go:104]: Yes, Once recommender is implemented we will save the data coming from listRecommendation API to ROS-OCP DB.

**Pull request merged**